UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs

```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
             return Unauthorized();
         }
 
+
+        [HttpGet]
+        [Produces("application/json")]
+        [Route("/ft/auth/unp")]
+        public IActionResult GetJwtAuthUnpTokenForFT([FromHeader(Name = "userPass")] string userPass, [FromHeader(Name = "Ocp-Apim-Subscription-Key")] string? subscriptionKey)
+        {
+            Dictionary<string, string> requestHeaders = new();
+            foreach (KeyValuePair<string, StringValues> header in Request.Headers)
+            {
+                requestHeaders.Add(header.Key, header.Value);
+            }
+            string? userName = _fleetManagerApiConfiguration.Value.UserName;
+            string? password = _fleetManagerApiConfiguration.Value.Password;
+            string? fleetManagerStubSubscriptionKey = _fleetManagerApiConfiguration.Value.SubscriptionKey;
+            string base64Credentials = GetBase64EncodedCredentials(userName, password);
+            HttpResponseMessage httpResponse = new();
+            string AuthToken = string.Empty;
+            if (string.IsNullOrEmpty(subscriptionKey))
+            {
+                string? invalidSubscriptionKeyResponse = "{\"statusCode\": 401, \"message\":\"Access denied due to missing subscription key. Make sure to include subscription key when making requests to an API.\"}";
+                httpResponse.StatusCode = HttpStatusCode.Unauthorized;
+                if (!httpResponse.IsSuccessStatusCode)
+                {
+                    return Unauthorized(JsonConvert.DeserializeObject<InvalidSubscriptionKeyResponse>(invalidSubscriptionKeyResponse));
+                }
+            }
+            if (subscriptionKey == fleetManagerStubSubscriptionKey)
+            {
+                if (userPass == base64Credentials)
+                {
+                    string? authTokenResponse = "{\"token\": \"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9\", \"expiration\":\"" + DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture) + "\"}";
```

Comment:
authTokenResponse doesn't need to be nullable.

Reply:
Changed